tests/src/kinAccount.unit.test.ts

```diff
@@ -285,61 +382,104 @@ describe("KinAccount.createAccount", async () => {
 			memoText: memo
 		});
 		txBuilder.addFee(20);
-		expect(await kinAccount.submitTransaction(txBuilder)).toEqual("708ebb9e3c8890333daca3faa6707b89dc0155f2578314e302f39e7387d2d07c");
+		expect(await kinAccount.submitTransaction(txBuilder)).toEqual(transactionId);
 	});
 
+	test("retry send kin, change fee before submitting transaction.", async () => {
+		mockLoadAccountResponse("6319125253062662");
+		mockRequest({ retry: 3, requestBody: bodySendKin });
+
+		const txBuilder = await kinAccount.buildSendKin({
+			address: receiverPublic,
+			amount: 23.3,
+			fee: amount,
+			memoText: memo
+		});
+		txBuilder.addFee(20);
+		expect(await kinAccount.submitTransaction(txBuilder)).toEqual(transactionId);
+		expect(retryDelay).toEqual(6);
+	});
+
+	test("retry send kin, error 500 ServerError.", async () => {
+		mockLoadAccountResponse("6319125253062662");
+		mockFailedRetries(6);
+
+		const txBuilder = await kinAccount.buildSendKin({
+			address: receiverPublic,
+			amount: 23.3,
+			fee: amount,
+			memoText: memo
+		});
+		txBuilder.addFee(20);
+		await expect(kinAccount.submitTransaction(txBuilder)).rejects.toEqual(ErrorDecoder.translate({response: mock500NetworkResponse}));
+	}, 120000);
 });
 
-function mock400SendKinResponse(response: ErrorResponse) {
-	nock(fakeUrl)
-		.matchHeader(headerKey, headerVal)
-		.post(url => url.includes("/transactions"), /tx=\w+/gi)
-		.reply(response.status, response);
+function mockFailedRetries(retries: number) {
+	const builder = nock(horizonUrl)
+		.matchHeader(headerKey, headerVal);
+	for (let i = 0; i < retries; i++) {
+		builder.post(url => true).reply(mock500NetworkResponse.status, mock500NetworkResponse);
+	}
 }
 
-function mock400AccountResponse(response: ErrorResponse) {
-	nock(fakeUrl)
+function mockResponse(response: ErrorResponse) {
+	nock(horizonUrl)
 		.matchHeader(headerKey, headerVal)
 		.post(url => url.includes("/transactions"), /tx=\w+/gi)
-		.reply(400, response);
+		.reply(response.status, response);
 }
 
-function mockLoadAccountResponse(sequence: string, url?: string) {
-	nock(url ? url : fakeUrl)
-		.matchHeader(headerKey, headerVal)
+function mockLoadAccountResponse(sequence: string, options?: { url?: string, retry?: number }) {
+	const builder = nock(options && options.url ? options.url : horizonUrl)
+		.matchHeader(headerKey, headerVal);
+	if (options && options.retry && options.retry > 0) {
+		for (let i = 0; i < options.retry; i++) {
```

Comment:
can use `nock.times()` instead

Reply:
Maybe our version is too far behind, the function does not exist on the scope.
@yosriz